service/cancel_orders_for_user.go

```diff
@@ -2,28 +2,41 @@ package service
 
 import (
 	"context"
-	"fmt"
 
 	"github.com/google/uuid"
 	"github.com/orbs-network/order-book/models"
 	"github.com/orbs-network/order-book/utils/logger"
 	"github.com/orbs-network/order-book/utils/logger/logctx"
 )
 
-func (s *Service) CancelOrdersForUser(ctx context.Context, userId uuid.UUID) (orderIds []uuid.UUID, err error) {
+func (s *Service) CancelOrdersForUser(ctx context.Context, userId uuid.UUID, symbol models.Symbol) (orderIds []uuid.UUID, err error) {
 
-	orderIds, err = s.orderBookStore.CancelOrdersForUser(ctx, userId)
-
-	if err == models.ErrNotFound {
+	ids, err := s.orderBookStore.GetOpenOrderIds(ctx, userId)
+	if err != nil {
 		logctx.Info(ctx, "no orders found for user", logger.String("userId", userId.String()))
 		return []uuid.UUID{}, err
 	}
-
-	if err != nil {
-		logctx.Error(ctx, "could not cancel orders for user", logger.Error(err), logger.String("userId", userId.String()))
-		return []uuid.UUID{}, fmt.Errorf("could not cancel orders for user: %w", err)
+	res := []uuid.UUID{}
+	for _, id := range ids {
+		order, err := s.getOrder(ctx, false, id)
```

Comment:
Use `FindOrdersByIds` here to only do one request to the DB

Reply:
done!